### PR TITLE
UCSC-164: FE Media Coverage block

### PR DESCRIPTION
**Pull request by bjcooper**
## What does this do/fix?

- Moves reusable component styles for pagination and button elements to the theme, which we're already closely coupled to it.
- Updated all coverage link CTA field label to be consistent with similar behavior in the Featured News block.
- Added classes, templating, and a bit of controller logic.
- Adds CTA support for this block in the query loop controller.
- **Requires updates that are going into the `ucsc-20221 theme**: https://github.com/ucsc/ucsc-2022/pull/360

### Links to relevant issues

- https://moderntribe.atlassian.net/browse/UCSC-164

### CODE REVIEW

File: src/Components/Query_Loop_Controller.php [hunk 1]
```diff
@@ -22,7 +23,9 @@ abstract protected function prepare_posts_for_display( array $posts = [], bool $
 	public function __construct( $block ) {
 		$this->block      = (array) $block;
 		$this->query_loop = (array) get_field( Query_Loop::QUERY_LOOP );
-		$this->cta        = (array) get_field( Featured_News_Block::CTA_FIELD );
+		$this->cta        = (array) get_field( Featured_News_Block::CTA_FIELD )
```
Comment (MlKilderkin) [inline src/Components/Query_Loop_Controller.php:26]: This should be moved from there to child class. E.g `Media_Coverage_Controller` will call 
```
public function __construct($block) {
        $this->cta = (array) get_field( Media_Coverage_Block::CTA_FIELD ) ?: [];
        
        parent::__construct($block);
    }
```
Same for Featured News. I missed on the last review but will have more blocks with that and will be completely unreadable

**Review by MlKilderkin — COMMENTED** [review]
🟡

File: src/Components/Media_Coverage_Controller.php [hunk 2]
```diff
@@ -22,9 +34,17 @@ public function get_title(): string {
...
 		return stripos( $url, $current_site ) !== false;
 	}
+	
+	public function get_cta(): string {
+		if ( empty( $this->cta ) || empty( $this->cta['title'] ) || empty( $this->cta['url'] ) ) {
+				return '';
+		}
+
+			return sprintf( '<a href="%s" class="wp-element-button" target="%s">%s</a>', $this->cta['url'], $this->cta['target'] ?: '_self', $this->cta['title'] );
+	}
```
[the remark this answers is not in the document]
Comment (bjcooper) [reply src/Components/Media_Coverage_Controller.php:47]: I need to add the `wp-element-button` class to the link in this case.

File: src/Components/Query_Loop_Controller.php (hunk 1 above)
Comment (bjcooper) [reply src/Components/Query_Loop_Controller.php]: Ah, thank you! Updated, please take another look.

**Review by MlKilderkin — APPROVED** [review]
🟢

**Pull request merged**